### PR TITLE
JAMES-3377 Text should match From field

### PR DESCRIPTION
**Pull request by chibenwa**
Also enhence address field matching as part of text field

### CODE REVIEW

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
`[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project james-server-mailets-integration-testing: There was a timeout or other error in the fork -> [Help 1]`

test this please

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailQueryMethodContract.scala [hunk 1]
```diff
@@ -4803,6 +4803,64 @@ trait EmailQueryMethodContract {
     }
   }
 
+  @Test
+  def textShouldMatchFromField(server: GuiceJamesServer): Unit = {
+    server.getProbe(classOf[MailboxProbeImpl]).createMailbox(inbox(BOB))
+    val messageId1: MessageId = server.getProbe(classOf[MailboxProbeImpl])
+      .appendMessage(BOB.asString, inbox(BOB), AppendCommand.from(Message.Builder
+        .of
```
Comment (LanKhuat) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailQueryMethodContract.scala:4811]: the query matches with the Body part, not the From?

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailQueryMethodContract.scala [hunk 2]
```diff
@@ -4861,7 +4861,7 @@ trait EmailQueryMethodContract {
          |    {
          |      "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
          |      "filter": {
-         |        "text": "test"
+         |        "text": "toto@domain.tld"
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailQueryMethodContract.scala:4864]: Fixup on the wrong test no?

**Comment by chibenwa** [comment]
test this please

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
test this please

**Review by nvduc91 — APPROVED** [review]

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**